#3 Import the required packages with their customary aliases

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#4 Use the `%matplotlib inline` magic command so that your plots show in the notebook _without_ having to call `plt.show()` every time.

In [2]:
%matplotlib inline

#5.	Using the pandas `read_csv()` method, read the GDP dataset into your notebook as a DataFrame called `gdp_df`. Take a look at the first 6 rows.

In [3]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv' )
gdp_df.head(6)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2020  1970.560169              NaN
1     Afghanistan  2019  2065.036235              NaN
2     Afghanistan  2018  2033.804389              NaN
3     Afghanistan  2017  2058.400221              NaN
4     Afghanistan  2016  2057.067978              NaN
5     Afghanistan  2015  2068.265904              NaN

#6. Repeat for the internet use dataset. Call this DataFrame `internet_df`. Take a look at the first six rows.


In [4]:
internet_df = pd.read_csv('../data/internet_use.csv')
internet_df.head(6)

Country or Area  Year     Value  Value Footnotes
0     Afghanistan  2014  6.390000              NaN
1     Afghanistan  2013  5.900000              NaN
2     Afghanistan  2012  5.454545              NaN
3     Afghanistan  2011  5.000000              NaN
4     Afghanistan  2010  4.000000              NaN
5     Afghanistan  2009  3.550000              NaN

#98. Look at the shape of each DataFrame - how many rows? How many columns?


In [5]:
gdp_df.shape

(7176, 4)

There are 7176 rows and 4 columns in gdp dataframe

In [6]:
internet_df.shape

(4676, 4)

There are 4675 rows and 4 columns in internet_use dataframe

#6.	Take a look at the datatypes for the columns in each DataFrame.


In [7]:
print(type(gdp_df['Country or Area']))
print(type(gdp_df.Year))
print(type(gdp_df.Value))
print(type(gdp_df['Value Footnotes']))

<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>


Show data types of columns:

In [8]:
gdp_df.columns

Index(['Country or Area', 'Year', 'Value', 'Value Footnotes'], dtype='object')

In [9]:
internet_df.columns


Index(['Country or Area', 'Year', 'Value', 'Value Footnotes'], dtype='object')

In [10]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7176 entries, 0 to 7175
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7176 non-null   object 
 1   Year             7176 non-null   int64  
 2   Value            7176 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 224.4+ KB


In [11]:
internet_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4676 entries, 0 to 4675
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  4676 non-null   object 
 1   Year             4676 non-null   object 
 2   Value            4495 non-null   float64
 3   Value Footnotes  948 non-null    float64
dtypes: float64(2), object(2)
memory usage: 146.2+ KB


Different datatypes between the two DFs. gdp_df: obj, int64, float64, float64. internet_df: object, object, float64, float64.

Drop Column 'Value Footnote'

#99. Take a look at the last 10 rows of each DataFrame in turn.

In [31]:
gdp_three.tail(10)

internet_three.tail(10)

Country                                               Year  \
4666     171                        Internet Dial-up customers.   
4667     172  Population age 16+ using the Internet in the l...   
4668     173  Population age 16+ using internet in the last ...   
4669     174  U.S. Census Bureau, Table 2. Reported Internet...   
4670     175             Includes individuals 3 years and older   
4671     176                                   NTIA/CPS survey.   
4672     177  Pew Internet Project, 'Internet, Broadband and...   
4673     178  Estimated based on Survey's results. Populatio...   
4674     179                     Preliminary. Country estimate.   
4675     180  The methodology used to estimated the figure f...   

      Internet_Users_Pct  
4666                 NaN  
4667                 NaN  
4668                 NaN  
4669                 NaN  
4670                 NaN  
4671                 NaN  
4672                 NaN  
4673                 NaN  
4674                 NaN  
4675                 NaN

#7.	Drop the `value footnotes` column from both DataFrames. Check that this worked as expected.


In [12]:
gdp_three = gdp_df.drop(columns = ['Value Footnotes'])
internet_three = internet_df.drop(columns = ['Value Footnotes'])
gdp_three.head()
internet_three.head()

Country or Area  Year     Value
0     Afghanistan  2014  6.390000
1     Afghanistan  2013  5.900000
2     Afghanistan  2012  5.454545
3     Afghanistan  2011  5.000000
4     Afghanistan  2010  4.000000

#8.	Change the columns for the GDP Per Capita DataFrame to ‘Country’, ‘Year’, and ‘GDP_Per_Capita’.


In [33]:
gdp_three.columns = ['Country','Year', 'GDP_Per_Capita']
gdp_three.head()


Country  Year  GDP_Per_Capita
0  Afghanistan  2020     1970.560169
1  Afghanistan  2019     2065.036235
2  Afghanistan  2018     2033.804389
3  Afghanistan  2017     2058.400221
4  Afghanistan  2016     2057.067978

#9.	Change the columns for the Internet Users DataFrame to ‘Country’, ‘Year’, and ‘Internet_Users_Pct’.

In [34]:
internet_three.columns = ['Country','Year', 'Internet_Users_Pct']
internet_three.head()

Country  Year  Internet_Users_Pct
0  Afghanistan  2014            6.390000
1  Afghanistan  2013            5.900000
2  Afghanistan  2012            5.454545
3  Afghanistan  2011            5.000000
4  Afghanistan  2010            4.000000

#10. Merge the two DataFrames to one. Merge **all rows** from each of the two DataFrames. Call the new DataFrame `gdp_and_internet_use`.

In [14]:
gdp_and_internet_use = pd.merge(gdp_three, internet_three, on=['Country'])
gdp_and_internet_use

Country  Year_x  GDP_Per_Capita Year_y  Internet_Users_Pct
0       Afghanistan    2020     1970.560169   2014            6.390000
1       Afghanistan    2020     1970.560169   2013            5.900000
2       Afghanistan    2020     1970.560169   2012            5.454545
3       Afghanistan    2020     1970.560169   2011            5.000000
4       Afghanistan    2020     1970.560169   2010            4.000000
...             ...     ...             ...    ...                 ...
108530     Zimbabwe    1990     2652.129209   1997            0.033080
108531     Zimbabwe    1990     2652.129209   1996            0.016790
108532     Zimbabwe    1990     2652.129209   1995            0.007684
108533     Zimbabwe    1990     2652.129209   1994            0.001739
108534     Zimbabwe    1990     2652.129209   1990            0.000000

[108535 rows x 5 columns]

#11. Look at the first five rows of your new DataFrame to confirm it merged correctly.


In [15]:
gdp_and_internet_use.head()


Country  Year_x  GDP_Per_Capita Year_y  Internet_Users_Pct
0  Afghanistan    2020     1970.560169   2014            6.390000
1  Afghanistan    2020     1970.560169   2013            5.900000
2  Afghanistan    2020     1970.560169   2012            5.454545
3  Afghanistan    2020     1970.560169   2011            5.000000
4  Afghanistan    2020     1970.560169   2010            4.000000

#12. Look at the last five rows to make sure the data is clean and as expected.

In [16]:
gdp_and_internet_use.tail()

Country  Year_x  GDP_Per_Capita Year_y  Internet_Users_Pct
108530  Zimbabwe    1990     2652.129209   1997            0.033080
108531  Zimbabwe    1990     2652.129209   1996            0.016790
108532  Zimbabwe    1990     2652.129209   1995            0.007684
108533  Zimbabwe    1990     2652.129209   1994            0.001739
108534  Zimbabwe    1990     2652.129209   1990            0.000000

#13. Subset the combined DataFrame to keep only the data for 2004, 2009, and 2014. Check that this happened correctly.

In [23]:
years_sub= gdp_and_internet_use['Year_x']
years_sub

0         2020
1         2020
2         2020
3         2020
4         2020
          ... 
108530    1990
108531    1990
108532    1990
108533    1990
108534    1990
Name: Year_x, Length: 108535, dtype: int64